### PR TITLE
[INLONG-7516][Manager][Sort][Agent] Decoupling Flink version dependencies for multiple versions of Apache Flink

### LINKED ISSUE #7516: [Improve][Manager][Sort][Agent] Decoupling Flink version dependencies for multiple versions of Apache Flink
### Description

Migrating Maven's version management decoupling Flink.

### InLong Component

InLong Manager, InLong Agent, InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
Changing Flink's version management.
1.My personal idea for this PR is to migrate Flink’s version management to each module, and the public dependencies will be placed under the root POM.
2.The following PR will extract common capabilities for different versions of Flink such as Sort-format, Sort-core, etc.
3.The final idea is to put the Flink version that Sort depends on in Sort-common, and only need to modify the POM of Sort-common when switching versions.

- Fixes #7516 

### Motivation

Decoupling Flink version dependencies for multiple versions of Apache Flink

### Modifications

Modified the POM

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,45 +40,74 @@
...
 |:-----------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
 | [![Stargazers over time](https://starchart.cc/apache/inlong.svg)](https://starchart.cc/apache/inlong) | [![Contributor Over Time](https://contributor-overtime-api.git-contributor.com/contributors-svg?chart=contributorOverTime&repo=apache/inlong)](https://git-contributor.com?chart=contributorOverTime&repo=apache/inlong) |
 
-[Apache InLong](https://inlong.apache.org) is a one-stop integration framework for massive data that provides automatic, secure and reliable data transmission capabilities. InLong supports both batch and stream data processing at the same time, which offers great power to build data analysis, modeling and other real-time  applications based on streaming data.
+[Apache InLong](https://inlong.apache.org) is a one-stop integration framework for massive data that provides automatic,
```
Comment (yunqingmoswu) [inline README.md:51]: Keep the origin style?

File: README.md [hunk 2]
```diff
@@ -40,45 +40,74 @@
...
+secure and reliable data transmission capabilities. InLong supports both batch and stream data processing at the same
+time, which offers great power to build data analysis, modeling and other real-time applications based on streaming
+data.
 
-InLong (应龙) is a divine beast in Chinese mythology who guides the river into the sea, and it is regarded as a metaphor of the InLong system for reporting data streams.
```
Comment (yunqingmoswu) [inline README.md:49]: Keep the origin style?

File: README.md [hunk 3]
```diff
@@ -40,45 +40,74 @@
...
+InLong (应龙) is a divine beast in Chinese mythology who guides the river into the sea, and it is regarded as a metaphor
+of the InLong system for reporting data streams.
 
-InLong was originally built at Tencent, which has served online businesses for more than 8 years, to support massive data (data scale of more than 80 trillion pieces of data per day) reporting services in big data scenarios. The entire platform has integrated 5 modules:  Ingestion, Convergence, Caching, Sorting, and Management, so that the business only needs to provide data sources, data service quality, data landing clusters and data landing formats, that is, the data can be continuously pushed from the source to the target cluster, which greatly meets the data reporting service requirements in the business big data scenario.
```
Comment (yunqingmoswu) [inline README.md:51]: Keep the origin style?

File: README.md [hunk 4]
```diff
@@ -40,45 +40,74 @@
...
+requirements in the business big data scenario.
 
 For getting more information, please visit our project documentation at https://inlong.apache.org/.
 ![inlong-structure-en.png](https://github.com/apache/inlong-website/blob/master/static/img/inlong-structure-en.png)
 
-
 ## Features
+
 Apache InLong offers a variety of features:
-* **Ease of Use**: a SaaS-based service platform. Users can easily and quickly report, transfer, and distribute data by publishing and subscribing to data based on topics.
```
Comment (yunqingmoswu) [inline README.md:59]: Keep the origin style?

File: README.md [hunk 5]
```diff
@@ -40,45 +40,74 @@
...
+* **Service Integration**: provides unified system monitoring and alert services. It provides fine-grained metrics to
+  facilitate data visualization. Users can view the running status of queues and topic-based data statistics in a
+  unified data metric platform. Users can also configure the alert service based on their business requirements so that
+  users can be alerted when errors occur.
+* **Scalability**: adopts a pluggable architecture that allows you to plug modules into the system based on specific
+  protocols. Users can replace components and add features based on their business requirements.
 
 ## When should I use InLong?
-InLong is based on MQ and aims to provide a one-stop, practice-tested module pluggable integration framework for massive data, based on this system, users can easily build stream-based data applications. It is suitable for environments that need to quickly build a data reporting platform, as well as an ultra-large-scale data reporting environment that InLong is very suitable for, and an environment that needs to automatically sort and land the reported data.
```
Comment (yunqingmoswu) [inline README.md:67]: Keep the origin style?

File: README.md [hunk 6]
```diff
@@ -40,45 +40,74 @@
...
 - Visualize your operations on [InLong dashboard](https://inlong.apache.org/docs/next/user_guide/dashboard_usage).
 
 ## Supported Data Nodes (Updating)
+
 | Type         | Name              | Version                      | Architecture          |
 |--------------|-------------------|------------------------------|-----------------------|
 | Extract Node | Auto Push         | None                         | Standard              |
 |              | File              | None                         | Standard              |
 |              | Kafka             | 2.x                          | Lightweight, Standard |
-|              | MongoDB           | >= 3.6                       | Lightweight, Standard |
```
Comment (yunqingmoswu) [inline README.md:80]: Keep the origin style?

File: README.md [hunk 7]
```diff
@@ -40,45 +40,74 @@
...
-|              | MQTT              | >= 3.1                       | Standard              |
```
Comment (yunqingmoswu) [inline README.md:81]: Keep the origin style?

File: README.md [hunk 8]
```diff
@@ -100,58 +129,81 @@ You can use InLong in the following ways：
 |              | PostgreSQL        | 9.6, 10, 11, 12              | Lightweight, Standard |
 |              | SQLServer         | 2012, 2014, 2016, 2017, 2019 | Lightweight, Standard |
 |              | TDSQL-PostgreSQL  | 10.17                        | Lightweight, Standard |
-|              | Doris             | >= 0.13                      | Lightweight, Standard |
```
Comment (yunqingmoswu) [inline README.md:103]: Keep the origin style?

File: README.md [hunk 9]
```diff
@@ -100,58 +129,81 @@ You can use InLong in the following ways：
...
-|              | StarRocks         | >= 2.0                       | Lightweight, Standard |
-|              | Kudu              | >= 1.12.0                    | Lightweight, Standard |
-|              | Redis             | >= 3.0                       | Lightweight, Standard |
+|              | Doris             | > = 0.13                      | Lightweight, Standard |
+|              | StarRocks         | > = 2.0                       | Lightweight, Standard |
+|              | Kudu              | > = 1.12.0                    | Lightweight, Standard |
+|              | Redis             | > = 3.0                       | Lightweight, Standard |
 
 ## Build InLong
-More detailed instructions can be found at [Quick Start](https://inlong.apache.org/docs/next/quick_start/how_to_build) section in the documentation.
```
Comment (yunqingmoswu) [inline README.md:109]: Keep the origin style?

**Comment by yunqingmoswu** [comment]
Please fix the unit test error. @GanfengTan

File: inlong-agent/pom.xml [hunk 10]
```diff
@@ -36,4 +36,54 @@
         <module>agent-release</module>
         <module>agent-docker</module>
     </modules>
+
+    <properties>
+        <debezium.version>1.8.0.Final</debezium.version>
```
Comment (healchow) [inline inlong-agent/pom.xml:41]: Is this modification related to the Flink version?

Comment (GanfengTan) [reply inlong-agent/pom.xml:41]: Each component depends on a different version of debezium. In the past, a version was managed under the root POM, but sort has its own version, which is very misleading.

File: inlong-sort/pom.xml [hunk 11]
```diff
@@ -80,6 +83,17 @@
...
+                <groupId>org.apache.flink</groupId>
+                <artifactId>flink-shaded-jackson</artifactId>
+                <version>${flink.jackson.version}</version>
+            </dependency>
+            <!--ojdbc8 is FUTC license, we use it test only-->
+            <dependency>
+                <groupId>com.oracle.database.jdbc</groupId>
+                <artifactId>ojdbc8</artifactId>
+                <version>${oracle.jdbc.version}</version>
+            </dependency>
```
Comment (gong) [inline inlong-sort/pom.xml:96]: It need add <scope>test</scope> ?

File: inlong-agent/pom.xml (hunk 10 above)
Comment (healchow) [reply inlong-agent/pom.xml:41]: Get it.

File: inlong-sort/pom.xml (hunk 11 above)
Comment (GanfengTan) [reply inlong-sort/pom.xml]: done

**Review by yunqingmoswu — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]
